pkg/hubble/parser/threefour/parser.go

```diff
@@ -430,16 +430,25 @@ func decodeICMPv6(icmp *layers.ICMPv6) *pb.Layer4 {
 	}
 }
 
-func decodeIsReply(tn *monitor.TraceNotify) *wrappers.BoolValue {
-	// Unfortunately, not all trace points have the connection
-	// tracking state available. For certain trace point
-	// events, we do not know if it actually was a reply or not.
-	if tn != nil && monitorAPI.TraceObservationPointHasConnState(tn.ObsPoint) {
+func decodeIsReply(tn *monitor.TraceNotify, pvn *monitor.PolicyVerdictNotify) *wrappers.BoolValue {
+	switch {
+	case tn != nil && monitorAPI.TraceObservationPointHasConnState(tn.ObsPoint):
+		// Unfortunately, not all trace points have the connection
+		// tracking state available. For certain trace point
+		// events, we do not know if it actually was a reply or not.
 		return &wrappers.BoolValue{
 			Value: tn.Reason & ^monitor.TraceReasonEncryptMask == monitor.TraceReasonCtReply,
 		}
+	case pvn != nil && pvn.Verdict >= 0:
+		// Forwarded PolicyVerdictEvents are emitted for the first packet of
+		// connection, therefore we statically assume that they are not reply
+		// packets
+		return &wrappers.BoolValue{Value: false}
+	default:
+		// For other events, such as drops, we simply do not know if they were
+		// replies or not.
+		return nil
```

Comment:
IIUC this function returns three possible values: "is a reply", "is not a reply", and "don't know", with "don't know" indicated by the `nil` return value.

Given that all callers currently call `.GetValue()` on the result which returns `false` if the result is `nil`, this means that "don't know" is currently always treated as "is not a reply".

It may be worth commenting this behavior at the call sites.

Reply:
> Given that all callers currently call `.GetValue()` on the result which returns false if the result is nil, this means that "don't know" is currently always treated as "is not a reply".

That's a good point, I'll add some comments. I would like to point out where it is important to distinguish between unknown and false, we _do_ check that `GetIsReply()` is not nil:

https://github.com/cilium/cilium/blob/0507c620b6a0fb0acc247a6276fcf4ef9c86bd3c/pkg/hubble/metrics/port-distribution/handler.go#L57

https://github.com/cilium/cilium/blob/2545bdde942a4f6b5f754242c8e90d3569679e8a/pkg/hubble/filters/reply.go#L35

The only cases where we completely and purposefully ignore the distinction between `reply=unknown` and `reply=false` is in the legacy `Flow.Reply` field.